webapp/src/tutorialCodeValidation.tsx

```diff
@@ -0,0 +1,63 @@
+/// <reference path="../../built/pxtlib.d.ts" />
+
+import * as React from "react";
+import * as data from "./data";
+import * as sui from "./sui";
+import { TutorialCard } from "./tutorial";
+
+type ISettingsProps = pxt.editor.ISettingsProps;
+
+interface TutorialCodeValidationProps extends ISettingsProps {
+    onYesButtonClick: () => void;
+    onNoButtonClick: () => void;
+    initialVisible: boolean;
+}
+
+interface tutorialCodeValidationState {
+    visible: boolean;
+}
+
+export class moveOn extends data.Component<TutorialCodeValidationProps, tutorialCodeValidationState> {
+    public elementRef: HTMLDivElement;
+    protected setRef: (el: HTMLDivElement) => void = (el) => { this.elementRef = el };
+
+    constructor(props: TutorialCodeValidationProps) {
+        super(props);
+
+        this.state = { visible: this.props.initialVisible };
+    }
+
+    showUnusedBlocksMessage(vis: boolean) {
+        this.setState({ visible: vis });
+    }
+
+    moveOnToNextTutorialStep() {
+        this.props.onYesButtonClick();
+        this.showUnusedBlocksMessage(false);
+    }
+
+    stayOnThisTutorialStep() {
+        this.props.onNoButtonClick();
+    }
+
+    renderCore() {
+        const vis = this.state.visible;
+        return <div id="tutorialCodeValidationID">
```

Comment:
small: don't need to add "ID" in the id. if this isn't being used in the CSS, you don't necessarily even need to specify an element id